tokio/src/stream/streamext/mod.rs

```diff
@@ -0,0 +1,190 @@
+use core::future::Future;
+use crate::stream::Stream;
+
+mod collect;
+mod filter;
+mod filter_map;
+mod map;
+mod next;
+
+#[allow(unreachable_pub)]
+pub use {
+    collect::Collect,
+    filter::Filter,
+    filter_map::FilterMap,
+    map::Map,
+    next::Next,
+};
+
+/// An extension trait for `Stream`s that provides a variety of convenient
+/// combinator functions.
+pub trait StreamExt: Stream {
+    /// Creates a future that resolves to the next item in the stream.
+    ///
+    /// Note that because `next` doesn't take ownership over the stream,
+    /// the [`Stream`] type must be [`Unpin`]. If you want to use `next` with a
+    /// [`!Unpin`](Unpin) stream, you'll first have to pin the stream. This can
+    /// be done by boxing the stream using [`Box::pin`] or
+    /// pinning it to the stack using the `pin_mut!` macro from the `pin_utils`
```

Comment:
I wonder if we should provide this macro too?

Reply:
We could consider adding additional utilities in a follow up.